// utils/apollo.js
import fetch from 'node-fetch';
import ApolloClient, { gql} from 'apollo-boost';

export default new ApolloClient({
	uri: "http://qkg7t3nssg.lb.c1.gra.k8s.ovh.net/v1/graphql",
	fetch,
});

export const  SERIES_CHAPTERS = gql`
query  {
  series_chapters(limit: 25, order_by: {time_uploaded: desc}) {
    id
    title
    chapter_number_volume
    chapter_number_absolute
    time_uploaded
    views
    comments
    series_chapters_series {
      id
      name
    }
    groups_series_chapters {
      groups_scanlation_series_chapters_groups {
        name
        id
      }
    }
  }
}
`;

export const SERIES = gql
	`
	query($id: Int!) {
	series(where: {id: {_eq: $id}}, limit: 1) {
		view
		updated
		status
		raw
		rating
		name
		mu_id
		mangadex_id
		mal_id
		id
		hentai
		follows
		description
		cover_image
		country
		bw_id
		amazonjp_id
series_chapters(limit: 25, order_by: {time_uploaded: desc}) {
    id
    title
    chapter_number_volume
    chapter_number_absolute
    time_uploaded
    views
    comments
    series_chapters_series {
      id
      name
    }
    groups_series_chapters {
      groups_scanlation_series_chapters_groups {
        name
        id
      }
    }
  }
		series_aliases {
		id
		name
		}
		people_series {
		people_series {
			id
			name
			type
		}
		}
		tags_series {
		tags_series {
			tag_name
			tag_namespace
		}
		}
		series_related {
			type
			series_related_related {
				id
				name
				cover_image
			}
		}
	}
	}

	`
